app/src/test/java/com/codesoom/assignment/auth/application/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,143 @@
+
+    @BeforeEach
+    void setUp() {
+        JwtTokenProvider jwtTokenProvider = new JwtTokenProvider(secret, EXPIRED_TIME);
+        ReflectionTestUtils.setField(jwtTokenProvider,
+                "time", LocalDateTime.of(2022, Month.JANUARY, 1, 0, 0));
+        authenticationService = new AuthenticationService(jwtTokenProvider, userRepository);
+
+        user = User.builder()
+                .id(GIVEN_ID)
+                .password(GIVEN_USER_PASSWORD)
+                .email(GIVEN_USER_EMAIL)
+                .build();
+    }
+
+    @Nested
+    @DisplayName("authenticate 메서드는")
+    class Describe_authenticate {
+
+        @Nested
+        @DisplayName("등록된 이메일과 비밀번호가 주어지면")
+        class Context_with_exist_email_password {
+            final String email = GIVEN_USER_EMAIL;
+            final String password = GIVEN_USER_PASSWORD;
+            final LoginRequest requestDto = new LoginRequest(email, password);
+
+            @BeforeEach
+            void setUp() {
+                given(userRepository.findByEmail(GIVEN_USER_EMAIL))
+                        .willReturn(Optional.of(user));
+            }
+
+            @DisplayName("인증 토큰을 리턴한다.")
+            @Test
+            void it_returns_token() {
+                String token = authenticationService.authenticate(requestDto);
+                assertThat(token).isEqualTo(VALID_TOKEN);
+            }
+        }
+
+        @Nested
+        @DisplayName("등록되지 않은 이메일이 주어지면")
+        class Context_with_not_exist_email {
+            final String email = NOT_EXIST_EMAIL;
+            final String password = GIVEN_USER_PASSWORD;
+            final LoginRequest requestDto = new LoginRequest(email, password);
+
+            @DisplayName("예외를 던진다.")
+            @Test
+            void It_throws_exception() {
+                assertThrows(IllegalArgumentException.class,
+                        () -> authenticationService.authenticate(requestDto));
+            }
+        }
+
+        @Nested
+        @DisplayName("유효하지않는 비밀번호가 주어지면")
+        class Context_with_wrong_password {
+            final String email = GIVEN_USER_EMAIL;
+            final String password = WRONG_PASSWORD;
+            final LoginRequest requestDto = new LoginRequest(email, password);
+
+            @BeforeEach
+            void setUp() {
+                given(userRepository.findByEmail(GIVEN_USER_EMAIL))
+                        .willReturn(Optional.of(user));
+            }
+
+            @DisplayName("예외를 던진다.")
+            @Test
+            void It_throws_exception() {
+                assertThrows(IllegalArgumentException.class,
+                        () -> authenticationService.authenticate(requestDto));
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("parseToken 메서드는")
+    class Describe_parseToken {
+        @Nested
+        @DisplayName("토큰이 주어지면")
+        class Context_with_token {
+            final String token = VALID_TOKEN;
+
+            @DisplayName("토큰에 담긴 정보를 리턴한다.")
```

Comment:
이렇게 바꾸면 더 정확할 것 같아요.
```suggestion
            @DisplayName("토큰에 담긴 정보를 가진 Claims를 리턴한다.")
```

Reply:
Claims 를 정보 조각으로 해야하나 고민을 했는데, 표현 그대로 하는 것도 좋은 방법이군요.

Reply:
주석에 클래스 이름을 그대로 넣는 건 저는 별로 선호하지 않지만, 항상 그런 것은 아닙니다 ㅎㅎ